streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollector.java

```diff
@@ -26,7 +26,13 @@
 public interface RecordCollector {
     <K, V> void send(ProducerRecord<K, V> record, Serializer<K> keySerializer, Serializer<V> valueSerializer);
 
-    <K, V> void send(ProducerRecord<K, V> record, Serializer<K> keySerializer, Serializer<V> valueSerializer,
+    <K, V> void send(final String topic,
```

Comment:
I think the first `send()` call can also be in refactored to not have `ProducerRecord` as its parameter: it is only used in the `StoreChangeLogger` which is also an internal class.